File: src/dynamicBody/character/enemy/Enemy.java
package dynamicBody.character.enemy;

import dynamicBody.bullet.Bullet;

/**
 * Interface that presents all the methods needed or that can be used with
 * regards to the enemy
 */
public interface Enemy extends dynamicBody.character.Character {

	/**
	 * Update the position of the enemy
	 */
	public void updatePos();

	/**
	 * Method use by Enemy.attack to put the bullet in the Set<Bullet>
	 * 
	 * @param bullet, the Bullet to add
	 */
	public void addBullet(Bullet bullet);

	/*
	 * Method use attack with the enemy
	 */
	public void attack();
	
	public TypeEnemy getType();

}

File: Ragazzini_Marco/CSharp_Task/DynamicBody/Character/Enemy/Enemy.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Model.DynamicBody.Character.Enemy
{
	public interface IEnemy : ICharacter
    {
		/// <summary>
		/// Update the position of the enemy
		/// </summary>
		public void UpdatePos();

	}
}
